### DISCUSSION TITLE
Is it possible to get types of `onupdate` and `ondelete` from `ForeignKeyConstraint` values as enum object?

### DISCUSSION THREAD

**Discussion by 0417taehyun** [sqlalchemy/sqlalchemy#9087] [Usage Questions] [answered]
I think it is little bit hard to set the type of foreign key constraint by typing manually.

Thus, I want to reuse the value like calling the enum object, such as `ForeignKeyConstrain.RESTRICT`.

Is it possible or do I need to make the enum object myself just like the one below?

```Python
import enum


class ForeignKeyConstraintType(str, enum.Enum):
    CASCADE = "CASCADE"
    RESTRICT = "RESTRICT"
    SET_NULL = "SET NULL"
```

**Comment by zzzeek** [comment, marked as answer]
we haven't moved to using enums for things in the public API as that's something which, if we do it, we'd want to do it all at once and not piecemeal.     That SQLA 2.0 uses Literal[] types for things like this (not the cascade type right now, but we could) means that IDEs should generally see the string names in any case the way it would an enum value.

For the FOREIGN KEY cascades though, that's more of a "SQL keyword" thing, which we tend to keep as strings, because some database backends might have some other odd value which for us is just a pass-through.  

for example, you put three enum values there.  There's also "NO ACTION" and "SET DEFAULT".  PostgreSQL lets you put [lists of column names there as well](https://www.postgresql.org/docs/current/sql-createtable.html) .  So for SQL inserts like ON UPDATE/ON DELETE CASCADE, enums are not the best option because the string is more open ended than an enum can allow.

**Reply by 0417taehyun** [reply]
Thanks for the detailed explanation @zzzeek !
I totally agree what you said. Thanks.
